File: Javascript_sol/Sliding_Window/Leetcode_219.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
 var containsNearbyDuplicate = function(nums, k) {
  let hmap = {};
  
  nums.forEach((ele, pos) => {
     if (hmap[ele]) {
         if ((pos - hmap[ele][0]) <= k) {
             k = -1
         };
         hmap[ele][0] = pos;
     } else {
         hmap[ele] = [pos, pos];
     };
  });
  
  if (k === -1) {return true}
  return false
};

File: Python_sol/Arrays_hashing/Leetcode_219.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hmap = {}

        for i, num in enumerate(nums):
            if hmap.get(num,-1) != -1 and i-hmap[num] <= k:
                return True
            hmap[num] = i